core/vdbe/execute.rs

```diff
@@ -1560,41 +1564,70 @@ pub fn op_column(
                                 data_slice[6],
                                 data_slice[7],
                             ];
-                            Value::Float(f64::from_be_bytes(bytes))
+                            Some(RefValue::Float(f64::from_be_bytes(bytes)))
                         } else {
-                            default.clone().unwrap_or(Value::Null)
+                            None
                         }
                     }
-                    n if n >= 12 && n % 2 == 0 => Value::Blob(data_slice.to_vec()),
-                    n if n >= 13 && n % 2 == 1 => Value::Text(Text {
-                        value: data_slice.to_vec(),
-                        subtype: TextSubtype::Text,
-                    }),
-                    _ => default.clone().unwrap_or(Value::Null),
+                    n if n >= 12 && n % 2 == 0 => {
+                        Some(RefValue::Blob(RawSlice::create_from(data_slice)))
+                    }
+                    n if n >= 13 && n % 2 == 1 => Some(RefValue::Text(TextRef::create_from(
+                        data_slice,
+                        TextSubtype::Text,
```

Comment:
Why RefValues? Isn't it possible we deallocate data_slice before using it with this refvalue?